src/utils.cpp

```diff
@@ -109,3 +130,163 @@ KeyRingReceive ZypperUtils::keyReport()
     return s_keyReceiveReport;
 }
 
+/************************************* ZYPPER INSTALL ************************************/
+void ZypperUtils::markPackagesForInstallation( const string& repoUrl, const string& packageName )
+{
+    Pathname sysRoot( "/" );
+    // acquire initial zypp lock
+    s_zypp = getZYpp(); 
+    
+    makeNecessaryChangesToRunAsRoot( sysRoot );
+    initTarget( sysRoot );
+    initRepoManager( repoUrl, packageName );
+    
+    //TRUST_AND_IMPORT_KEY - we are adding this to the system repos
+    KeyRing::setDefaultAccept( KeyRing::TRUST_AND_IMPORT_KEY );
+    cout << "TRUST_AND_IMPORT_KEY accepted" << endl;
+    refreshRepoManager();
+    
+    //load cache
+    try {
+	cout << "Loading resolvables from " << s_repoInfo.alias() << endl;
+	s_repoManager->loadFromCache( s_repoInfo );      
+    }
+    catch (const Exception& exp) {
+	// cachefile has old format or is corrupted: try to rebuild it
+	s_repoManager->cleanCache( s_repoInfo );
+	s_repoManager->buildCache( s_repoInfo );
+	s_repoManager->loadFromCache( s_repoInfo );
+    }
+    
+    // select/specify package(s) to install
+    PoolQuery q;
+    q.addKind( ResKind::package );
+    q.addAttribute( sat::SolvAttr::name, packageName );
+    q.setMatchExact();
+    
+    PoolItemBest bestMatches(q.begin(), q.end());    
+    if ( !bestMatches.empty() ) {
+	for ( PoolItemBest::iterator it = bestMatches.begin(); it != bestMatches.end(); ++it ) {
+	    ui::asSelectable()( *it )->setOnSystem( *it, ResStatus::USER );
+	}
+    }
```

Comment:
I admit I'm not familiar with one-click installer and it's intended workflow and I did not dig deeper into the code. But this query caught my eye: I assume one-click installer should install a package from a _specific(!) temp. repo_, but neither the PoolQuery nor the filter lop reflect this constraint. 

Maybe because this _specific repo_ is the only one loaded into the pool and so you don't expect additional matches?
If this is true, it's IMO a bad idea:

1) Without the enabled repositories, you won't be able to resolve missing dependencies of your package. 
Might be undesired, but does not do harm to the system.

2) Installed packages which are not available in any repository are considered to be `orphaned` (unwanted, dropped, no longer supported,.. however you call it). If `orphaned` packages are involved in a dependency conflict, the resolver may consider to __silently remove__  them to resolve the conflict. `Wanted` packages however (those also mentioned in  available repos) will never be deleted without raising the conflict. 
Enabled repos do not just offer packages, they also kind of protect their installed packages. You probably don't want ot install without having them in the pool too.